cdk/domino_cdk/config/base.py

```diff
@@ -127,6 +128,26 @@ def val(path: str, obj):
                     errors.append(f"{path}.{f.name} type ({f.type}) does not match value: [{value}] ({type(value)})")
 
         val("config", self)
+
+        def last_az(azs: str):
+            return sorted(azs, reverse=True)[0]
+
+        def az_pos(azs: str):
+            if azs:
+                return ascii_lowercase.index(last_az(azs)[-1]) + 1
+            return 0
+
+        for ng, cfg in self.eks.managed_nodegroups.items():
+            if az_pos(cfg.availability_zones) > self.vpc.max_azs:
+                errors.append(
+                    f"Nodegroup {ng} has availability zone {last_az(cfg.availability_zones)} exceeding vpc.max_azs count of {self.vpc.max_azs}"
+                )
+        for ng, cfg in self.eks.unmanaged_nodegroups.items():
+            if az_pos(cfg.availability_zones) > self.vpc.max_azs:
+                errors.append(
+                    f"Nodegroup {ng} has availability zone {last_az(cfg.availability_zones)} exceeding vpc.max_azs count: {self.vpc.max_azs}"
+                )
```

Comment:
this isn't right.
All you care about is that the AZs selected match the AZs we have defined in the VPC.
We can check that the number of AZs selected is less than or equal to the VPC AZs but not really necessary.

Reply:
so an easier way is to just do `len(set(cfg.availability_zones)) > self.vpc.max_azs` no need to do anything fancy.

Reply:
I switched it up so it looks at what AZs are in the configured region, replicates the logic of the vpc, and checks if each nodegroup's AZs are in the projected VPC's subnet AZs. This should cover regions where an AZ letter is skipped.